pkg/local_object_storage/engine/get.go

```diff
@@ -54,99 +54,81 @@ func (e *StorageEngine) Get(addr oid.Address) (*objectSDK.Object, error) {
 
 func (e *StorageEngine) get(addr oid.Address, shardFunc func(s *shard.Shard, ignoreMetadata bool) (hasMetadata bool, err error)) error {
 	var (
-		ok    bool
-		siErr *objectSDK.SplitInfoError
-
-		errNotFound apistatus.ObjectNotFound
-
-		outSI    *objectSDK.SplitInfo
-		outError error = errNotFound
-
+		hasDegraded   bool
 		shardWithMeta shardWrapper
+		splitInfo     *objectSDK.SplitInfo
 		metaError     error
 	)
 
-	var hasDegraded bool
-	var objectExpired bool
-
-	e.iterateOverSortedShards(addr, func(_ int, sh shardWrapper) (stop bool) {
+	for _, sh := range e.sortedShards(addr) {
 		noMeta := sh.GetMode().NoMetabase()
 		hasDegraded = hasDegraded || noMeta
 
 		hasMetadata, err := shardFunc(sh.Shard, noMeta)
 		if err != nil {
+			var siErr *objectSDK.SplitInfoError
+
 			if hasMetadata {
 				shardWithMeta = sh
 				metaError = err
 			}
 			switch {
 			case shard.IsErrNotFound(err):
-				return false // ignore, go to next shard
+				continue // ignore, go to next shard
 			case errors.As(err, &siErr):
-				if outSI == nil {
-					outSI = objectSDK.NewSplitInfo()
+				if splitInfo == nil {
+					splitInfo = objectSDK.NewSplitInfo()
 				}
 
-				util.MergeSplitInfo(siErr.SplitInfo(), outSI)
+				util.MergeSplitInfo(siErr.SplitInfo(), splitInfo)
 
 				// stop iterating over shards if SplitInfo structure is complete
-				return !outSI.GetLink().IsZero() && !outSI.GetLastPart().IsZero()
+				if !splitInfo.GetLink().IsZero() && !splitInfo.GetLastPart().IsZero() {
+					return logicerr.Wrap(objectSDK.NewSplitInfoError(splitInfo))
```

Comment:
That's in fact one of the things I wanted to discuss. _It looks_ like split info collection we attempt to do here (with iteration-persistent `splitInfo` is totally useless, we can return immediately upon seeing _any_ split info because that's what we do ultimately anyway. But maybe I'm missing something.

Reply:
split information may or may not have a link object and a last part (it is theoretically ok to have a middle object only, be resynced, etc)

Reply:
Regarding the comment above --- having a complete split info is beneficial to the caller (even though it gets an error), so this `if` can't be removed even though lacking a proper info we will return whatever is available.

Regarding break/return --- both are valid, so it doesn't matter much, post-loop can be extended in which case `return` is better even though currently it duplicates error wrapping a bit.